packages/gatsby-theme-styleguide/src/gatsby-components/SiteMapData/tools.js

```diff
@@ -0,0 +1,69 @@
+/*
+basedir = '/src/docs' - using allFile
+basedir = 'docs' - using allMdx
+*/
+
+export const parseAllFiles = (list, { basedir = 'docs/', location } = {}) => {
+  const finalArray = []
+
+  // find dir obj
+  const findDirObj = (array, path) => {
+    return array.find(elem => elem.path === path)
+  }
+
+  list.forEach(elem => {
```

Comment:
in those loops and functions can we relate to the files in more precise terms : 
list => allMdx
elem => mdx or mdxFile
in ((array, path) =>)
array => ?? what is it ? splitPathArray ?
path ? same here

findDirObj => what does finDirObj ? is it the right name for this function ?